pytorch_lightning/trainer/trainer.py

```diff
@@ -2026,6 +2026,21 @@ def should_rank_save_checkpoint(self) -> bool:
     def num_nodes(self) -> int:
         return getattr(self.strategy, "num_nodes", 1)
 
+    @property
+    def device_ids(self) -> List[int]:
+        devices = getattr(self.strategy, "parallel_devices", [self.strategy.root_device])
+        device_ids = []
+        for idx, device in enumerate(devices):
+            if isinstance(device, torch.device):
+                device_ids.append(device.index or idx)
+            elif isinstance(device, int):
+                device_ids.append(device)
+        return device_ids
+
+    @property
+    def num_devices(self) -> int:
+        return len(self.device_ids)
```

Comment:
can it rely on `trainer._accelerator_connector.devices` here? since we are only concerned with the count, and not the actual devices. `device_ids` does some additional minor computation.

Reply:
I am planning to remove devices from `AcceleratorConnector` since that's not internally used by `AcceleratorConnector` and it's not meant to be a public property. 

But I could definitely implement the same logic here so the computations are more efficient.

Reply:
Moved the logic here. Also in tests checked they are the same with len(device_ids)